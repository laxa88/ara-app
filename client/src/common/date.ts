// https://stackoverflow.com/questions/23593052/format-javascript-date-to-yyyy-mm-dd

const getDateData = () => {
  const d = new Date();

  const year = d.getFullYear();
  let month = `${d.getMonth() + 1}`;
  let day = `${d.getDate()}`;

  if (month.length < 2) {
    month = `0${month}`;
  }
  if (day.length < 2) {
    day = `0${day}`;
  }

  return { year, month, day };
};

/**
 * Returns current month, e.g. "2018-11"
 */
const getMonth = () => {
  const { year, month } = getDateData();
  const resultDate = [year, month].join('-');
  return resultDate;
};

/**
 * Returns today's date, e.g. "2018-11-16"
 */
const getDate = () => {
  const { year, month, day } = getDateData();
  const resultDate = [year, month, day].join('-');
  return resultDate;
};

/**
 * Returns current time, e.g. "23:42:09"
 */
const getTime = () => {
  const d = new Date();

  let h = d.getHours().toString();
  let m = d.getMinutes().toString();
  let s = d.getSeconds().toString();

  if (h.length < 2) {
    h = `0${h}`;
  }

  if (m.length < 2) {
    m = `0${m}`;
  }

  if (s.length < 2) {
    s = `0${s}`;
  }

  const resultTime = [h, m, s].join(':');

  return resultTime;
};

/**
 * Returns formatted date, e.g. "2018-11-16 23:42:09"
 */
const formatDate = () => {
  const result = `${getDate()} ${getTime()}`;
  return result;
};

/**
 * Parses data, e.g. "2018-12-01T15:00:00.000Z" to "2018-12-01"
 */
const parseDate = (date: string) => {
  return date.substr(0, 10);
};

export { getMonth, getDate, getTime, formatDate, parseDate };
