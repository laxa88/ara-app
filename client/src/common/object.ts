const getValue = (object: any, propName: string) => {
  const arr = propName.split('.');
  let result = object;

  while (arr.length && result) {
    result = result[arr.shift()];
  }

  return result;
};

export { getValue };
