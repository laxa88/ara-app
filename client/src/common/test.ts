import { configure, mount, render, shallow } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import reduxMockStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

// Reserved for later on, if we use thunk or saga or anything else
const middleware: any = [];

function mockStore(state: any) {
  return reduxMockStore(middleware)(state);
}

export { shallow, mount, render, mockStore };
