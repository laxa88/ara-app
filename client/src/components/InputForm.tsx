import * as React from 'react';

import Form from './Form';
import Input, { IProps as IInput } from './Input';

export interface IProps extends IInput {
  label: string;
}

class InputForm extends React.Component<IProps, {}> {
  constructor(props: IProps) {
    super(props);
  }

  public render() {
    const { label, value, onChange, ...others } = this.props;

    return (
      <Form label={label}>
        <Input value={value} onChange={onChange} {...others} />
      </Form>
    );
  }
}

export default InputForm;
