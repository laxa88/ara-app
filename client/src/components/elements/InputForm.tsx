import * as React from 'react';
import styled from '../css/styled';

import Input, { IInput } from './Input';

const Label = styled.label`
  color: red;
`;

export interface IProps extends IInput {
  label: string;
}

class InputForm extends React.Component<IProps, {}> {
  constructor(props: IProps) {
    super(props);
  }

  public render() {
    const { label, value, onChange, ...others } = this.props;

    return (
      <form>
        <Label>{label}</Label>
        <Input value={value} onChange={onChange} {...others} />
      </form>
    );
  }
}

export default InputForm;
