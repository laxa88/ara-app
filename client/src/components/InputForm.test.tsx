import * as React from 'react';
import { shallow } from '../common/test';
import InputForm from './InputForm';

describe('<InputForm />', () => {
  it('should render', () => {
    const mockProps = {
      label: 'dummy label',
      onChange: jest.fn(),
      value: 'dummy value',
    };

    const wrapper = shallow(<InputForm {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
