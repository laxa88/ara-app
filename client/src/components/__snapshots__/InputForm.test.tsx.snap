// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm /> should render 1`] = `
<Form
  label="dummy label"
>
  <Input
    onChange={[MockFunction]}
    value="dummy value"
  />
</Form>
`;
